src/main/java/com/morlunk/jumble/net/JumbleSSLSocketFactory.java

```diff
@@ -89,7 +92,11 @@ public SSLSocket createSocket(InetAddress host, int port) throws IOException {
      * @return The remote server's certificate chain, or null if a connection has not reached handshake yet.
      */
     public X509Certificate[] getServerChain() {
-        return mTrustWrapper.getServerChain();
+        if(mTrustWrapper instanceof JumbleTrustManagerWrapper) {
```

Comment:
Perhaps it may be a good idea to make an interface for a trust manager that can provide the remote's certificate chain. instanceof is generally discouraged.